AirCasting/CoreData/MeasurementStreamStorage.swift

```diff
@@ -31,13 +32,20 @@ final class CoreDataMeasurementStreamStorage: MeasurementStreamStorage {
 
     private let persistenceController: PersistenceController
     private lazy var updateSessionParamsService = UpdateSessionParamsService()
-
+    
+    let context: NSManagedObjectContext
+    
     init(persistenceController: PersistenceController) {
         self.persistenceController = persistenceController
+        self.context = persistenceController.editContext()
     }
-
+    
+    func save() throws {
+        try context.save()
```

Comment:
this should really be dispatched to the context queue (like basically everything public in this class. We have no real control over what thread this func can be called from and NSManagedContexts are really thread sensitive)